packages/sdk/base/src/async.ts

```diff
@@ -139,3 +139,29 @@ export async function concurrentValuesMap<IN extends any, OUT extends any>(
     return output
   }, {})
 }
+
+export const timeout = <T extends any[], U>(
+  inFunction: InFunction<T, U>,
```

Comment:
Shouldn't this be a Promise?

Reply:
It could be..I guess it felt cleaner to create the promise inside the timeout function. It's also what the other functions here are doing and the code pattern was easier to copy. Let me know if you feel strongly about this